reporter/urlparams.js

```diff
@@ -0,0 +1,96 @@
+( function() {
+
+// Only interact with URLs via window.location
+var location = typeof window !== "undefined" && window.location;
+if ( !location ) {
+	return;
+}
+
+var urlParams = getUrlParams();
+
+QUnit.urlParams = urlParams;
+
+// Match module/test by inclusion in an array
+QUnit.config.moduleId = [].concat( urlParams.moduleId || [] );
+QUnit.config.testId = [].concat( urlParams.testId || [] );
```

Comment:
This leaves in place the current sloppiness (privileged parameters are applied immediately and unconditionally, while `urlConfig` parameters wait for runStart and only overwrite `undefined` values), but I think it's fine to save that cleanup for another ticket and commit (cf. https://github.com/jquery/qunit/issues/955#issuecomment-198474631 ).